airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/S3Destination.java

```diff
@@ -45,10 +50,52 @@ public AirbyteConnectionStatus check(final JsonNode config) {
 
   @Override
   public AirbyteMessageConsumer getConsumer(final JsonNode config,
-                                            final ConfiguredAirbyteCatalog configuredCatalog,
-                                            final Consumer<AirbyteMessage> outputRecordCollector) {
+      final ConfiguredAirbyteCatalog configuredCatalog,
+      final Consumer<AirbyteMessage> outputRecordCollector) {
     final S3WriterFactory formatterFactory = new ProductionWriterFactory();
     return new S3Consumer(S3DestinationConfig.getS3DestinationConfig(config), configuredCatalog, formatterFactory, outputRecordCollector);
   }
 
+  public static void attemptS3WriteAndDelete(final S3DestinationConfig s3Config, final String bucketPath) {
+    final var prefix = bucketPath.isEmpty() ? "" : bucketPath + (bucketPath.endsWith("/") ? "" : "/");
+    final String outputTableName = prefix + "_airbyte_connection_test_" + UUID.randomUUID().toString().replaceAll("-", "");
+    attemptWriteAndDeleteS3Object(s3Config, outputTableName);
+  }
+
+  private static void attemptWriteAndDeleteS3Object(final S3DestinationConfig s3Config, final String outputTableName) {
+    final var s3 = getAmazonS3(s3Config);
+    final var s3Bucket = s3Config.getBucketName();
+
+    s3.putObject(s3Bucket, outputTableName, "check-content");
+    s3.deleteObject(s3Bucket, outputTableName);
+  }
+
+  public static AmazonS3 getAmazonS3(final S3DestinationConfig s3Config) {
```

Comment:
would a unit test make sense here?

Reply:
added _a_ test, but it's pretty mediocre. `AmazonS3` objects don't actually expose any methods to extract their configurations, so all the test accomplishes is to check that this method doesn't throw an exception, and that it returns something non-null :/

Reply:
actually, I can just kill this method and use `S3DestinationConfig#getS3Client` instead